src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -867,15 +863,19 @@ private List<GradingInstruction> copyGradingInstruction(GradingCriterion origina
      */
     public void validateDates() {
         // All fields are optional, so there is no error if none of them is set
-        if (getReleaseDate() == null && getDueDate() == null && getAssessmentDueDate() == null && getExampleSolutionPublicationDate() == null) {
+        if (getReleaseDate() == null && getStartDate() == null && getDueDate() == null && getAssessmentDueDate() == null && getExampleSolutionPublicationDate() == null) {
             return;
         }
         if (isExamExercise()) {
             throw new BadRequestAlertException("An exam exercise may not have any dates set!", getTitle(), "invalidDatesForExamExercise");
         }
 
         // at least one is set, so we have to check the three possible errors
-        boolean areDatesValid = isNotAfterAndNotNull(getReleaseDate(), getDueDate()) && isValidAssessmentDueDate(getReleaseDate(), getDueDate(), getAssessmentDueDate())
+        boolean areDatesValid = isNotAfterAndNotNull(getReleaseDate(), getDueDate()) && isNotAfterAndNotNull(getReleaseDate(), getStartDate())
+                && isNotAfterAndNotNull(getStartDate(), getDueDate()) //
+                && isValidAssessmentDueDate(getStartDate(), getDueDate(), getAssessmentDueDate()) //
+                && isValidAssessmentDueDate(getReleaseDate(), getDueDate(), getAssessmentDueDate()) //
```

Comment:
I assume these comments are to stop the autoformatting? I think it would be better to use `// @formatter:off` and `// @formatter:on` here.

Also move the `isNotAfterAndNotNull(getReleaseDate(), getStartDate())` to a new line to be consistent.

Reply:
I agree. I'll look at it later